### DISCUSSION TITLE
The correct place to put the security authorization boundary for the system and list the IP addresses (CIDR) blocks and FQDNs for the host in the SSP

### DISCUSSION THREAD

**Discussion by emili910** [usnistgov/OSCAL#1073] [Q&A] [answered]
{Please enter your question.}
Per David's note opening a discussion topic on the following question:

Where in the SSP is the correct place to put the security authorization boundary for the system? More precisely, where is the right place to list all the IP addresses (CIDR blocks) for the network, the FQDNs of the hosts? In some larger systems, this could be a challenge, i.e., listing all the resources. For Cloud – it is even greater contest because with ELB and deployment of resources on demand capabilities. Thanks

**Comment by iMichaela** [comment]
@emili910 - The list of IPs will be associated with the inventory items. For example, if you have 10 Linux servers, each one will be an inventory item and the IP addresses will be listed in the inventory. This is under: `/system-implementation/system-inventory/inventory-item/prop[@name="ipv4-address"]/@value`
or `/system-implementation/system-inventory/inventory-item/prop[@name="ipv6-address"]/@value`
If a device has multiple IP addresses, then include one `prop` field for each IP address.

**Comment by aj-stein-nist** [comment]
Michaela did a good job explaining how the inventory would work (FYI, there is no more intermediary `system-inventory` in XML or JSON, `inventory-item`s are directly under `system-implementation` [per the current docs](https://pages.nist.gov/OSCAL/reference/latest/system-security-plan/json-definitions/#/assembly/oscal-ssp/system-implementation/inventory-items)). 

The one thing I am not sure we can help you with, yet, is how you asked a few times to describe where in OSCAL to put "the security authorization boundary for the system" because how you describe that very thing. FYI, the authorization boundary is meant to be a prose description, customarily accompanied by a diagram illustrating the boundary described in its description. [The OSCAL model documentation for `system-security-plan` points this out](https://pages.nist.gov/OSCAL/reference/latest/system-security-plan/json-definitions/#/assembly/oscal-ssp/authorization-boundary/description). It does _not_ include IP addressing and other relevant network metadata, just a description of the network topology and design _without_ explicitly including that data. I do not think that is a FedRAMP style, that seems to be generally trust of non-FedRAMP SSPs I had seen in the past as well.

Can you give an idea of what is challenging you with the authorization boundary portion of a SSP?

**Comment by iMichaela** [comment]
@emili910 and @aj-stein-nist -- The `authorization boundary` diagram can be included in the `back-mater` and the diagram can list the IPs, if desired by the company. With that said, I don't think a diagram would include inventory items because it might become overly crowded and unreadable. I hope Emil is not thinking of IP addresses of system interconnections and interfaces.

**Comment by emili910** [comment]
Thank you for the detailed description Michaela and AJ. The challenge is that in the majority of the SSPs (the hard copy ones - non OSCAL format) and for FedRAMP, the inventory is listed usually in the SSP-A13-FedRAMP-Integrated-Inventory-Workbook-Template.xlsx. Correct me if I am wrong. But for systems that are non FedRAMP - I have seen many examples of incomplete info everywhere. I was just trying to parallel OSCAL (machine format) against the human-readable and he details provided are great help. 
For the boundary diagram - indeed it could become a very crowded if we list the IPs
I was not thinking about the IP addresses for the sys interconnections and interfaces.

**Comment by GaryGapinski** [comment]
System (security plan) components are either inside (subject to authorization) or outside (not subject to authorization but of interest) the authorization boundary (a 2-manifold give or take a dimension) and a component's inclusion in the boundary must be explicit (or inferred from a component-to-component transitive relationship). Authorization boundary membership (inclusion or exclusion) is necessarily an attribute of each component (if one expects to "draw" a boundary).

A "diagram" (graph) delineating authorization boundary inclusion would employ component boundary inclusion attributes.

I expect a separate discussion of the difference between a _component_ and an _inventory item_ will be interesting. Both are either insiders or outsiders: but an inventory item might be an instance of a component, a component, less than or not a component, or something somehow so different than a component that it either can not nor should not be expressed as a component.

**Comment by aj-stein-nist** [comment]
> Correct me if I am wrong. But for systems that are non FedRAMP - I have seen many examples of incomplete info everywhere. I was just trying to parallel OSCAL (machine format) against the human-readable and he details provided are great help. For the boundary diagram - indeed it could become a very crowded if we list the IPs I was not thinking about the IP addresses for the sys interconnections and interfaces.

So to circle back on this @emili910 you have a good sense of what you need to complete a system inventory, generic or FedRAMP variety, or you still need more concrete examples?

**Comment by emili910** [comment, marked as answer]
I am good thanks @aj-stein-nist . Thank you all for your help!

**Comment by iMichaela** [comment]
The NIST RMF team's reply to this issue:

The authorization boundary establishes the scope of protection for the system (i.e., what the organization agrees to protect under its management control).  The term “system” is used to define the set of system elements/system components, and system element interconnections [for more information see, SP 800-37, Rev 2, Task P-11, Section 2.5 and Appendix G]. Additionally, [SP 800-53] control PL-2, System Security and Privacy Plans provides additional guidance on the content of said plans. **If a system component is included in the system security plan, it is inherently part of the system and in the authorization boundary.**
 
Ultimately, the **system security plan (SSP) captures implementation of the selected controls and how the controls are applied to different system components (within the authorization boundary)**; **NIST does not and will not prescribe “the correct place to put the security authorization boundary.**” SSPs often include narrative descriptions and other references such as network diagrams, security architectures, etc. that capture the authorization boundary and scope of the system. An implementing organization may elect to include information about IP address blocks, etc. to describe the scope of the authorization boundary; the suite of NIST Risk Management Framework guidance and OSCAL schema allows for flexibility in implementation.
 
Regarding your comment about “component” and “inventory item,” refer to SP 800-37, Section 2.4 “System and System Elements” and [SP 800-53] control CM-8, system component inventory.  This control, which is selected for all security control baselines, details the need to develop, document, review/update/maintain an inventory of system components (i.e., discrete, identifiable IT assets that include hardware, software, and firmware).
